### DISCUSSION TITLE
How to correctly dispose DevToolsContext?

### DISCUSSION THREAD

**Discussion by n303ef** [cefsharp/CefSharp.Dom#40] [Q&A] [answered]
When form with browser is closed, very often I get `System.ObjectDisposedException`: 

"This instance of CefSharp.IBrowserHost been disposed!"

It occurs in CefBrowserHostWrapper.cpp method `CefBrowserHostWrapper::SendDevToolsMessage` and messge is:

`"{\"id\":70,\"method\":\"Runtime.disable\"}"`

In method `BrowserForm_FormClosing` I tried such calls:

```
DevToolsContext.DisposeAsync();
ChromiumWebBrowser.Dispose();
IBrowser.Dispose();
```

None of them in any combination (or even without any of these) helped.

**Comment by amaitland** [comment, marked as answer]
Upgrade to [2.0.83](https://www.nuget.org/packages/CefSharp.Dom/2.0.83)

If you are going to dispose the `ChromiumWebBrowser` instance then you don't need to `Dispose` the `DevToolsContext`.

See #41

**Reply by n303ef** [reply]
Thanks, upgrade to 2.0.83 helped.
